src/tactic/ui/widget/data_export_wdg.py

```diff
@@ -1328,48 +1461,72 @@ def get_display(my):
             var server = TacticServerStub.get();
             var class_name = 'pyasm.command.CsvImportCmd';
 
-            var response_div = bvr.src_el.getParent('.spt_panel').getElement('.spt_cmd_response');
+            var response_div = src_el.getParent('.spt_panel').getElement('.spt_cmd_response');
+            var csv_control = src_el.getParent('.spt_panel').getElement('.spt_csv_sample')
+            
+            var run_cmd;
+
             var on_complete = function(response_div) {
+                spt.app_busy.hide();
                 
                 response_div.innerHTML = 'CSV Import Completed';
                  
                 setTimeout( function() { 
-                    
                     var popup = bvr.src_el.getParent(".spt_popup");
                     if (popup) {
                         spt.popup.hide_background();
                         popup.destroy();
                     }
-                    
-                    spt.table.run_search() } , 2000);
-
+                    spt.table.run_search() } , 2000
+                );
             }
-            spt.app_busy.show("Importing Data");
            
-            var has_error = false;
-            var src_el = bvr.src_el;
-
+            /* On error includes handling of failure for single commit.
+             * If a single commit fails, user is prompted to skip that 
+             * entry or abort entire import. */
             var on_error = function(e) {
                 var err_message = spt.exception.handler(e);
-                spt.error(err_message);
+
+                spt.app_busy.hide();
                 
-                var response_div = document.getElement('.spt_cmd_response');
+                // This function is executed when Command fails or user
+                // chooses to abort after single entry fails.
+                var abort = function() {
+                    var response_div = document.getElement('.spt_cmd_response');
+                    err_message = err_message.replace(/\\n/g,'<br/>');
+                    response_div.innerHTML = 'Error: ' + err_message;
+                    response_div.setStyle("display", "");
+                    spt.show(csv_control);
+                }
 
-                err_message = err_message.replace(/\\n/g,'<br/>');
-                response_div.innerHTML = 'Error: ' + err_message;
-                response_div.setStyle("display", "");
-                setTimeout(function() {spt.show(csv_control)}, 500);
-           }
-            var csv_control = src_el.getParent('.spt_panel').getElement('.spt_csv_sample')
+                if (err_message.startsWith("Error creating new entry for row [")) {
```

Comment:
this Error creating new entry for row  should be a class constant.. so if the message changes in the future , it is only changed once